packages/custom-esbuild/src/custom-esbuild-schema.ts

```diff
@@ -0,0 +1,9 @@
+import { ApplicationBuilderOptions, DevServerBuilderOptions } from '@angular-devkit/build-angular';
+
+export type CustomEsbuildApplicationSchema = ApplicationBuilderOptions & {
+  plugins?: string[];
+};
+
+export type CustomEsbuildDevServerSchema = DevServerBuilderOptions & {
+  middlewares?: string[];
```

Comment:
If possible let's generate these with `quicktype` like [here](https://github.com/just-jeb/angular-builders/blob/master/packages/timestamp/package.json#L29), so that we have `.json` as a single source of truth for the schema.